xarray/structure/concat.py

```diff
@@ -100,14 +108,18 @@ def concat(
             load the data payload of data variables into memory if they are not
             already loaded.
           * "all": All data variables will be concatenated.
+          * None: Means ``"all"`` if ``dim`` is not present in any of the ``objs``,
```

Comment:
This is new.

We want to distinguish between concatenating ålong an existing dimension (numpy's concatenate) and concatenating along a new dimension (numpy's stack).

Without this (and with `data_vars="minimal", coords="minimal"` as default), we are not doing any concatenation if asked to concatenate along a new dim. That behaviour is certainly wrong. So now the new default we aremigrating to is `data_vars=None, coords="minimal"` where we will at least stack the data vars.